TelCo.ColorCoder/NumberToColor.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+    class NumberToColor{
+
+         /// <summary>
+        /// Given the two colors the function returns the pair number corresponding to them
+        /// </summary>
+        /// <param name="pair">Color pair with major and minor color</param>
+        /// <returns></returns>
+        public static int GetPairNumberFromColor(ColorPair pair)
+        {
+            // Find the major color in the array and get the index
+            int majorIndex = -1;
+            for (int i = 0; i < ColorMap.colorMapMajor.Length; i++)
+            {
+                if (ColorMap.colorMapMajor[i] == pair.majorColor)
+                {
+                    majorIndex = i;
+                    break;
+                }
+            }
+
+            // Find the minor color in the array and get the index
+            int minorIndex = -1;
+            for (int i = 0; i < ColorMap.colorMapMinor.Length; i++)
```

Comment:
See comment above - ColorMap.colorMapMinor has the colormap repeating twice - not required you can call it ColorMap.MinorColor for example